File: lab3/s1/ejb/src/main/java/ru/uvuv643/ejb/remote/DatabaseConfig.java
package ru.uvuv643.ejb.remote;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class DatabaseConfig {
    private static final String URL = "jdbc:postgresql://postgres:5432/your_database";
    private static final String USER = "your_user";
    private static final String PASSWORD = "your_password";

    static {
        try {
            Class.forName("org.postgresql.Driver");
        } catch (ClassNotFoundException e) {
            throw new IllegalStateException("PostgreSQL JDBC Driver not found. Include it in your library path.", e);
        }
    }

    public static Connection getConnection() throws SQLException {
        return DriverManager.getConnection(URL, USER, PASSWORD);
    }
}

File: lab2/s1/src/main/kotlin/ru/uvuv643/soa/server/DatabaseConfig.kt
package ru.uvuv643.soa.server

import java.sql.Connection
import java.sql.DriverManager
import java.util.Properties

object DatabaseConfig {
    private val url: String
    private val user: String
    private val password: String

    init {

        try {
            Class.forName("org.postgresql.Driver") // Ensure the driver is loaded
        } catch (e: ClassNotFoundException) {
            throw IllegalStateException("PostgreSQL JDBC Driver not found. Include it in your library path.", e)
        }

        val props = Properties()
        val resource = this::class.java.classLoader.getResource("application.properties")
            ?: throw IllegalStateException("Could not locate application.properties in resources folder")

        props.load(resource.openStream())

        url = props.getProperty("db.url")
        user = props.getProperty("db.user")
        password = props.getProperty("db.password")
    }

    fun getConnection(): Connection {
        return DriverManager.getConnection(url, user, password)
    }
}